content/reference/crosswalk-aws/lambda.md

```diff
@@ -0,0 +1,375 @@
+---
+title: "AWS Lambda and Serverless Events"
+menu:
+  reference:
+    parent: crosswalk-aws
+    name: Lambda
+    weight: 9
+---
+
+[AWS Lambda](https://aws.amazon.com/lambda/) lets you run code without provisioning or managing servers. You pay only
+for the compute time you consume and there is no charge when your code is not running. With Lambda, you can run code
+for virtually any type of application or backend service with zero administration. Just upload your code and Lambda
+takes care of everything required to run and scale your code with high availability. You can set up your code to
+automatically trigger from other AWS services or call it directly from any web or mobile app.
+
+{{< mini-toc >}}
+
+## Overview
+
+Pulumi Crosswalk for AWS brings a more natural, and easier to use, way of building serverless applications using
+AWS Lambda. Pulumi lets you express Lambda functions using real code, and handles packaging, versioning, and
+configuration of the associated AWS resources. This lets you focus on your application logic without needing to
+worry about boilerplate, and with confidence that the resulting infrastructure automatically uses AWS best practices.
+
+With Pulumi Crosswalk for AWS, event sources are available on all native resource types, including AWS S3, SQS,
+DynamoDB, CloudWatch, Kinesis, and more, in addition to full support for [API Gateway]({{< relref "api-gateway.md" >}}).
+This improves discoverability of event sources in addition to adding strong typing to the event handler
+inputs and outputs that AWS Lambda will deliver to your code.
+
+## Available AWS Services with Event Sources
+
+Below is a list of AWS Services and their available Lambda event sources:
+
+<style>
+table {
+    margin-bottom: 20px;
+    font-size: 14px;
+}
+
+td, th {
+    padding: 8px 8px;
+    text-align: left;
+    border: 1px solid rgba(0,0,0,0.13);
+}
+
+thead tr th {
+    background-color: #e0e0e0;
+    font-weight: 800;
+}
+
+thead tr th:first-child {
+    text-align: left;
+}
+
+tbody tr td:first-child {
+    text-align: left;
+}
+</style>
+
+| AWS Service | Event | Description |
+|-------------|-------|-------------|
+| API Gateway | [awsx.apigateway.API](/reference/pkg/nodejs/@pulumi/awsx/apigateway/#API) | create serverless APIs using a simple approach |
```

Comment:
```suggestion
| API Gateway | [awsx.apigateway.API]({{< relref "/reference/pkg/nodejs/pulumi/awsx/apigateway#API" >}}) | create serverless APIs using a simple approach |
```

Two comments (applies throughout):

1. The URL to the reference docs should not include the `@` in `@pulumi`. Hugo strips the `@` in the published directory structure, so the URLs are now `/reference/pkg/nodejs/pulumi/...`. We do have redirects on pulumi.io that redirect from `@pulumi` to `pulumi` based routes, and will have redirects that go from pulumi.io to their new homes on www.pulumi.com when we flip the migration switch, but we aren't planning to keep around the redirects within www.pulumi.com.

2. Also, I've been leveraging the [`relref` shortcode](https://gohugo.io/content-management/shortcodes/#ref-and-relref) for links within the site. With that, Hugo will ensure the page being linked to exists, and if not, will fail the build, which is especially helpful as we shift content around in subdirs as part of the website migration. (It also allows linking directly to `foo.md` pages, so you don't have to guess what the published URL is going to be).

Reply:
Re (1), yeah, I found out the hard way about this. (I was using `#` anchors, and they were getting stripped during the redirection.) I'll do a pass to clean these up -- thanks for catching!

Re (2), ahh, ok, I will see about changing all the URLs I've got in there to adopt this.

Reply:
Done, thanks again :+1: